compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/DesktopPopup.desktop.kt

```diff
@@ -44,52 +51,97 @@ fun rememberCursorPositionProvider(
     alignment: Alignment = Alignment.BottomEnd,
     windowMargin: Dp = 4.dp
 ): PopupPositionProvider = with(LocalDensity.current) {
-    val component = LocalLayerContainer.current
-    val cursorPoint = remember {
-        val awtMousePosition = MouseInfo.getPointerInfo().location
-        convertPointFromScreen(awtMousePosition, component)
-        IntOffset(
-            (awtMousePosition.x * component.density.density).toInt(),
-            (awtMousePosition.y * component.density.density).toInt()
-        )
-    }
+    val windowMarginPx = windowMargin.roundToPx()
     val offsetPx = IntOffset(offset.x.roundToPx(), offset.y.roundToPx())
+    val cursorPosition = rememberCursorPosition()
+
+    PopupPositionProviderAtOffset(
+        anchorPx = cursorPosition,
+        isAnchorRelative = false,
+        offsetPx = offsetPx,
+        alignment = alignment,
+        windowMarginPx = windowMarginPx
+    )
+}
+
+/**
+ * A [PopupPositionProvider] that positions the popup at the given position relative to the anchor.
+ *
+ * @param positionPx the offset, in pixels, relative to the anchor, to position the popup at.
+ * @param offset [DpOffset] to be added to the position of the popup.
+ * @param alignment The alignment of the popup relative to desired position.
+ * @param windowMargin Defines the area within the window that limits the placement of the popup.
+ */
+@Composable
+fun popupPositionProviderAtPosition(
+    positionPx: IntOffset,
```

Comment:
We mix Dp and Px here. We should keep only Px or only Dp. I suggest Dp, because it can be used in a high-level UI code.

Reply:
But `positionPx` comes from the cursor position. It would be strange to have to convert it from px to dp to use this function.
I can change the other parameter to dp, but that would also make it a bit less convenient to use.
Is it a real problem to mix px and dp?

Reply:
We should think about all use cases of this API, not only about our own. There is no reason other than our use case to have different units here. For other cases it adds inconvinience and inconsistency.

If we just keep px here, better to remove this function entirely, and only expose the class itself.

Reply:
I agree with the premise, but I think this **is** a useful method to provide for users because it's a common use-case to say "I want the put the popup at _this pixel position_ (the mouse/press event position) but offset it by _this dp position_ (to move away from the mouse pointer/finger).

Also, I think I will make it `remember` the returned value. The most common use-case is when this is used in the composition, and there's no point re-creating the popup provider every recomposition.

Reply:
You can see this use-case in `TooltipArea`.

Reply:
Ack